### PR TITLE
Provide minSequenceNr for snapshot deletion

### PR DESCRIPTION
**Pull request by ismaelhamed**
Port [#25590](https://github.com/akka/akka/pull/25590)

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs [hunk 1]
```diff
@@ -99,21 +99,22 @@ protected override bool ReceiveCommand(object message)
...
-                    Log.Warning("PersistentShard snapshot failure: {0}", m.Cause.Message);
+                    Log.Warning("PersistentShard snapshot failure: [{0}]", m.Cause.Message);
                     break;
                 case DeleteMessagesSuccess m:
-                    Log.Debug("PersistentShard messages to {0} deleted successfully", m.ToSequenceNr);
-                    DeleteSnapshots(new SnapshotSelectionCriteria(m.ToSequenceNr - 1));
+                    var deleteTo = m.ToSequenceNr - 1;
+                    var deleteFrom = Math.Max(0, deleteTo - Settings.TunningParameters.KeepNrOfBatches * Settings.TunningParameters.SnapshotAfter);
+                    Log.Debug("PersistentShard messages to [{0}] deleted successfully. Deleting snapshots from [{1}] to [{2}]", m.ToSequenceNr, deleteFrom, deleteTo);
+                    DeleteSnapshots(new SnapshotSelectionCriteria(deleteTo, DateTime.MaxValue, deleteFrom));
```
Comment (ismaelhamed) [inline src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs]: maxTimeStamp is not required in the JVM. Are we missing a SnapshotSelectionCriteria overload here?

**Comment by Aaronontheweb** [comment]
For some reason CI never ran on this branch.... No idea why

**Comment by Aaronontheweb** [comment]
Ah, old commit date - September 29th.

File: src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs]: @ismaelhamed I think so - mind submitting a separate PR for that?

**Pull request merged**